docs/docsite/rst/user_guide/complex_data_manipulation.rst

```diff
@@ -0,0 +1,205 @@
+        name: '{{ namestuff|reject('equal_to', omit)|list }}'
+      state: absent
+      vars:
+        namestuff:
+            # here we make this option 'omittable' by assigning the 'omit' token on a condition
+          - "{{ (fs_installed_smb_v1|default(False))|ternary(omit, 'FS-SMB1') }}"
+          - "foo"
+          - "bar"
+
+
+Another way is to avoid adding elements to the list in the first place, so you can just use it directly:
+
+.. code::
+
+    - name: build unique list with some items conditionally omittted
+      set_fact:
+         namestuff: ' {{ (namestuff|default([]))|union([item]) }}'
+      when: item != omit
+      loop:
+          - "{{ (fs_installed_smb_v1|default(False))|ternary(omit, 'FS-SMB1') }}"
+          - "foo"
+          - "bar"
+
+
+.. _complex_type_transfomations:
+
+Complex Type transformations
+=============================
+
+Jinja provides filters for simple data type tranformations (int, bool, etc), but when you want to tranform data structures things are not as easy.
+You can use the loops and list comprehensions as shown above to help, also other filters and lookups can be chained and leverage to achieve these comlex transformations.
+
+
+.. _create_dictionary_from_list:
+
+Create dictionary from list
+---------------------------
+
+In most languages it is easy to create a dictionary (aka map/associatve array/hash/etc) from a list of pairs, in Ansible there are a couple of ways to do it and the best one for you might depend on the source of your data.
+
+
+These example produces ``{"a": "b", "c": "d"}``
+
+.. code::
+
+  vars:
+      single_list: [ 'a', 'b', 'c', 'd' ]
+      mydict: "{{ dict(singlelist) }}"
+
+
+A bit more complex, using ``set_fact`` and a ``loop`` to create/update a dictionary with key value pairs from 2 lists:
+
+.. code::
+
+     - name: Uses 'combine' to update the dictionary and 'zip' to make pairs of both lists
+       set_fact:
+         mydict: "{{ mydict|default({}) | combine({item[0]: item[1]}) }}"
+       loop: "{{ (keys| zip(values))|list }}"
+       vars:
+         keys:
+           - foo
+           - var
+           - bar
+         values:
+           - a
+           - b
+           - c
+
+This results in ``{"foo": "a", "var": "b", "bar": "c"}``.
+
+
+You can even combine these simple examples with other filters and looups to create a dictionary dynamically by matching patterns to variable names:
+
+.. code::
+
+    vars:
+      myvarnames: "{{q('varnames', '^my')}}"
+      mydict: "{{ dict(myvarnames | zip(q('vars', \*myvarnames))) }}"
+
+A quick explanation, since there is a lot to unpack from those 2 lines
+
+ - The 'varnames' lookup returns a list of variables that matches 'begin with "my"'
+ - When feedng the list from the prevuois step into the 'vars' lookup to get the list of values.
+   The ``*`` is used to 'dereference the list' (a pythonism that works in Jinja), otherwise it would take the list as one argument.
```

Comment:
```suggestion
   The ``*`` is used to 'dereference the list' (a pythonism that works in Jinja), otherwise it would take the list as a single argument.
```